src/test/java/core/basesyntax/dao/StorageDaoImplTest.java

```diff
@@ -0,0 +1,85 @@
+package core.basesyntax.dao;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertNull;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.Fruit;
+import java.util.HashMap;
+import java.util.Map;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class StorageDaoImplTest {
+    private static final Integer AMOUNT = 100;
+    private static final String FRUIT_NAME = "banana";
+    private static final String NAME_OF_NOT_EXISTING_FRUIT = "grape";
+    private static StorageDao storageDao;
+    private static Map<Fruit, Integer> expectedMap;
+    private Fruit fruit;
+
+    @BeforeAll
+    static void beforeAll() {
+        storageDao = new StorageDaoImpl();
+        expectedMap = new HashMap<>();
+    }
+
+    @BeforeEach
+    void setUp() {
+        fruit = new Fruit(FRUIT_NAME);
+        expectedMap.put(fruit, AMOUNT);
+    }
+
+    @Test
+    void storageDao_add_isOk() {
+        storageDao.add(fruit, AMOUNT);
+        Map<Fruit, Integer> actualMap = Storage.fruits;
+        assertEquals(expectedMap, actualMap);
+    }
+
+    @Test
+    void storageDao_getExistingFruit_isOk() {
+        Storage.fruits.put(fruit, AMOUNT);
+        Map.Entry<Fruit, Integer> actualEntry = storageDao.get(fruit.getName());
+        Fruit actualFruit = actualEntry.getKey();
+        Integer actualAmount = actualEntry.getValue();
+        assertEquals(fruit, actualFruit);
+        assertEquals(AMOUNT, actualAmount);
+    }
+
+    @Test
+    void storageDao_getNullIfNotExist_isOk() {
+        Storage.fruits.put(fruit, AMOUNT);
+        Map.Entry<Fruit, Integer> actualEntry = storageDao.get(NAME_OF_NOT_EXISTING_FRUIT);
+        assertNull(actualEntry);
+    }
+
+    @Test
+    void storageDao_getAll_isOk() {
+        Storage.fruits.put(fruit, AMOUNT);
+        Map<Fruit, Integer> actualMap = storageDao.getALl();
+        assertEquals(expectedMap, actualMap);
+    }
+
+    @Test
+    void storageDao_isInStorage_isOk() {
+        Storage.fruits.put(fruit, AMOUNT);
+        boolean actual = storageDao.isInStorage(fruit.getName());
+        assertTrue(actual);
+    }
+
+    @Test
+    void storageDao_isNotInStorage_isOk() {
+        boolean actual = storageDao.isInStorage(FRUIT_NAME);
+        assertFalse(actual);
+    }
```

Comment:
Try to divide statements in tests by the following principle:
https://medium.com/@gitaeklee/given-when-then-junit-test-ba49564303e7

writing comments like //when //then is optional; you can divide everything by adding empty lines between statements